src/structs.rs

```diff
@@ -83,49 +83,60 @@ impl Mergeable for Value {
   }
 }
 
-#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, PartialOrd, Ord, Eq, Hash)]
-pub struct EdgeType(String);
 
-impl EdgeType {
+pub trait EdgeType {
+  fn is_alias_from(&self) -> bool;
 
-  pub const TO_RIGHT: &str = ":->";
-  pub const FROM_LEFT: &str = ":<-";
-  pub const CONTAINS_DOWN: &str = ":\\/";
-  pub const CONTAINED_BY_UP: &str = ":^";
+  fn is_alias_to(&self) -> bool;
 
-  pub const CONTAINED_BY: &str = "ContainedBy:^";
-  pub const CONTAINS: &str = "Contains:\\/";
-  pub const ALIAS_TO: &str = "AliasTo:->";
-  pub const ALIAS_FROM: &str = "AliasFrom:<-";
+  fn is_from_left(&self) -> bool;
 
+  fn is_to_right(&self) -> bool;
 
-  pub fn is_alias_from(&self) -> bool {
-    self.0 == EdgeType::ALIAS_FROM
+  fn is_contains_down(&self) -> bool;
+
+ fn is_contained_by_up(&self) -> bool;
+}
+
+pub const TO_RIGHT: &str = ":->";
+pub const FROM_LEFT: &str = ":<-";
```

Comment:
Thinking about this last night, what's the difference between 'up' and 'left', 'down' and 'right' ?

I get the sense that you're thinking of package names as the same 'level' in the hierarchy as the gitoid, but in strictly graph terms, it's above it / needs one more traversal in either case.

Reply:
Aliases are Left/Right in my head. A purl or md5 hash that points to a record via an alias is Left/Right. These records are "peers"

Contains (i.e., a JAR file contains .class files) and ContainedBy (a .class file is ContainedBy a JAR file). These records are hierarchical.

Reply:
Makes sense, if not in a graph theoretic way, in an information hierarchy way.

Reply:
In some ways it's doing what "To" and "From" already do, though that doesn't give a hierarchy.